vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp

```diff
@@ -602,6 +657,13 @@ void TextInputShadowNode::updateProperties(winrt::Microsoft::ReactNative::JSValu
   }
 
   Super::updateProperties(props);
+
+  // We need to re-register the PreviewKeyDown handler so it is invoked after the ShadowNodeBase handler
+  if (hasKeyDownEvents) {
```

Comment:
Isn't Subclass::UpdateProperties called after the `ShadowNodeBase` gets bound in the `PreviewKeyboardEventHandlerOnRoot` constructor?

Reply:
Yes - this is necessary. Although the root-level keyboard event handler for emitting key events gets subscribed before any of these props get set, the view-level keyboard event handler that sets `Handled(true)` on the routed events gets subscribed in `Super::updateProperties`.

If we subscribe to `PreviewKeyDown` for `onSubmitEditing` before we subscribe to `PreviewKeyDown` for `keyDownEvents`, then we'll always call `onSubmitEditing`, even if the key was intended to be marked Handled by the `keyDownEvents` prop.

Note, this wasn't a problem before because you could just use `keyDownEvents={[{code: 'Enter', handledEventPhase: Capturing}]}` to mark the event as Handled in the preview phase, and the KeyDown event that was previously used for `onSubmitEditing` would not be called.

Reply:
Got it. Then, I have a few questions:
1. Since `UpdateProperties` will be called at least once (and potentially many times during the lifespan of the component), do we need still need the initial registration up in `registerEvents()`? Not sure how expensive these are, though..
2. This seems to result in the re-registration of the Preview handler every time that `UpdateProperties` is called for a TextInput with `keyDownEvents` specified - can we avoid that?
3. Just to confirm - do you need to switch from listening to `KeyDown` to `PreviewKeyDown,` so that you can stop the submission on 'Enter'?

Reply:
1. We still need the initial registration in case keyDownEvents is never set.
2. I don't believe it's that expensive, but it will only be re-registered any time someone changes keyDownEvents, which is likely to be infrequent. It's probably not worth the added complexity to add a state field that signals when we've already re-registered.
3. Not exactly. We now need to handle onSubmitEditing in PreviewKeyDown as opposed to KeyDown because we want submitKeyEvents to apply even to TextBox with AcceptsReturn == true. If we used KeyDown, there would be no way to intercept 'Enter'.

Reply:
> If we subscribe to PreviewKeyDown for onSubmitEditing before we subscribe to PreviewKeyDown for keyDownEvents, then we'll always call onSubmitEditing, even if the key was intended to be marked Handled by the keyDownEvents prop.

This relies on the event handlers being called in the order of their registration - is that guaranteed (and documented) somewhere?